governance/api-reviews.md

```diff
@@ -0,0 +1,205 @@
+
+Including the name “experimental” in an API endpoint allows you to break
+backwards compatibility in the future, as noted above. However, we still prefer
+to mark the experimental API as deprecated for one release before removing it in
+the subsequent release. Please do not use the experimental namespace as an
+escape hatch for bad code or functionality you
+don’t-really-want-but-need-for-now; experimental APIs should be APIs that we
+intend to make standard in the near future, but have some lingering questions to
+resolve first.
+
+
+### Docstrings 
+
+TF APIs should have comprehensive documentation in the form of docstrings. If at
+all possible these docstrings should have runnable examples, and these examples
+should form a doctest so they stay correct. The examples should demonstrate an
+end-to-end user workflow, such that it’s clear how to generate the necessary
+inputs for the API and what to do with the outputs. The docstring should be
+understandable by someone who is not familiar with TF. See the [guide to writing
+TF docstrings](https://www.tensorflow.org/community/contribute/docs_ref) for
+more information.
+
+Our documentation generator for classes only sees methods, so prefer defining
+members as properties instead of assigning them in `__init__`.
+
+Docstrings should only refer to other public TF API symbols (i.e. do not refer
+to other symbols defined in the same file as a function which is just now being
+made public) and should refer to public API symbols by their full exported name.
+
+### Common names
+
+
+Prefer keepdims over keep_dims. Prefer axis over dim. Data types are called
+dtype. name is a common last argument of ops but backward compatibility mandates
+that new arguments are added after the last existing argument, even if that
+results in name not being the last argument.
+
+We generally prefer spelling things out over using abbreviations except when
+abbreviations are more standard than spelling things out (i.e. don’t spell out
+linalg or svd). When in doubt we ask domain experts or use web search to see
+what spelling is most common.
+
+If possible we prefer to name things in a similar way to numpy (e.g., we would
+not pick einsum as a name, but numpy and MatLab before it has, and that
+precedent is very strong).
+
+We prefer experimental namespaces (i.e. tf.data.experimental.foobar) over
+experimental-prefixed names (i.e. tf.data.experimental_foobar) except when
+adding an experimental class method, or an experimental argument. Experimental
+endpoints should be deprecated in a minor release before they can be removed in
+the next. We would like new experimental symbols to be things which will
+eventually end up in core TF as opposed to things we expect will be phased out
+with no clear replacement. The best expectation to have for an experimental
+endpoint is that the “experimental” will simply be removed. If you don’t believe
+that’ll work, it should probably not be added in its current form.  Style
```

Comment:
Should `Style` here be a different section header?

Reply:
Oops.